internal/app/subsystems/api/grpc/grpc_test.go

```diff
@@ -23,10 +24,11 @@ import (
 
 type grpcTest struct {
 	*test.API
-	subsystem api.Subsystem
-	errors    chan error
-	conn      *grpc.ClientConn
-	client    grpcApi.PromiseServiceClient
+	subsystem      api.Subsystem
+	errors         chan error
+	conn           *grpc.ClientConn
+	client         grpcApi.PromiseServiceClient
+	scheduleClient grpcApi.SchedulesClient
```

Comment:
At first, I liked that the promise and schedule clients were separate. But the tests made me realize that the user would also need to instantiate a different client for schedules :(

Do you think think we should combine into one client? Maybe there is a `protoc` command we could use to combine multiple files into a single client and that way we can still keep the files separated (which would be nice for syncing with the durable promises specification repo).

Reply:
this is exactly how we do the cli currently. since we have one ClientSet with multiple clients. 

i'm not sure the context a user instantiate this client directly. but from the sdk we can do a clientSet with clients like cli.